src/enumerable.rb

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    length.times do |i|
+      yield(self[i])
+    end
+  end
+
+  def my_each_with_index
+    length.times do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    selected_items = []
+    my_each do |item|
+      selected_items << item if yield(item)
+    end
+    selected_items
+  end
+
+  def my_all?
+    my_each { |item| return false unless yield(item) }
+    true
+  end
+
+  def my_any?
+    my_each { |item| return true if yield(item) }
+    false
+  end
+
+  def my_none?
+    my_each { |item| return false if yield(item) }
+    true
+  end
+
+  def my_count
+    count = 0
+    my_each { |item| count += 1 if yield(item) }
+    count
+  end
+
+  # 11 - Modify your #my_map method to take a proc instead.
+  # def my_map(proc)
+  #   result = []
+  #   my_each { |item| result << proc.call(item) }
+  #   result
+  # end
+
+  def my_map
+    result = []
+    my_each { |item| result << yield(item) }
+    result
+  end
+
+  def my_inject(acc = 0)
```

Comment:
#### 19.  should be able to combine all the elements of a collection with a given operation:

```
irb(main):095:0> (1..2).inject { |acc, n| acc + n }
=> 3
irb(main):096:0> (1..2).inject(1) { |acc, n| acc + n }
=> 4
irb(main):097:0> (1..2).inject(10) { |acc, n| acc + n }
=> 13
irb(main):098:0> (1..2).inject(100) { |acc, n| acc + n }
=> 103
```

#### 20. Should be able to perform such operation is passed a symbol as an argument:

```
irb(main):099:0> (1..2).inject(:+)
=> 3
irb(main):103:0> (2..4).inject(:*)
=> 24
```